x-pack/plugins/ml/public/embeddables/anomaly_swimlane/anomaly_swimlane_embeddable_factory.tsx

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { StartServicesAccessor } from '@kbn/core/public';
+import { KibanaContextProvider } from '@kbn/kibana-react-plugin/public';
+import { KibanaThemeProvider } from '@kbn/react-kibana-context-theme';
+import fastIsEqual from 'fast-deep-equal';
+
+import type { ReactEmbeddableFactory } from '@kbn/embeddable-plugin/public';
+import { initializeTimeRange, initializeTitles } from '@kbn/presentation-publishing';
+import React, { Suspense, useEffect } from 'react';
+import { BehaviorSubject, Subscription } from 'rxjs';
+import type { AnomalySwimlaneEmbeddableServices, AnomalySwimlaneEmbeddableUserInput } from '..';
+import { ANOMALY_SWIMLANE_EMBEDDABLE_TYPE } from '..';
+import type { MlDependencies } from '../../application/app';
+import type { SwimlaneType } from '../../application/explorer/explorer_constants';
+import { SWIM_LANE_DEFAULT_PAGE_SIZE } from '../../application/explorer/explorer_constants';
+import { HttpService } from '../../application/services/http_service';
+import type { MlPluginStart, MlStartDependencies } from '../../plugin';
+import type { JobId } from '../../shared';
+import { buildDataViewPublishingApi } from '../common/anomaly_detection_embeddable';
+import { EmbeddableLoading } from '../common/components/embeddable_loading_fallback';
+import EmbeddableSwimLaneContainer from './embeddable_swim_lane_container';
+import type { AnomalySwimLaneEmbeddableApi, AnomalySwimLaneEmbeddableState } from './types';
+
+/**
+ * Provides the services required by the Anomaly Swimlane Embeddable.
+ */
+export const getServices = async (
+  getStartServices: StartServicesAccessor<MlStartDependencies, MlPluginStart>
+): Promise<AnomalySwimlaneEmbeddableServices> => {
+  const [coreStart, pluginsStart] = await getStartServices();
+
+  const { AnomalyDetectorService } = await import(
```

Comment:
nit: it seems like these 5 awaits could be parallelized with `Promise.all`. That might speed this up a lot.

Reply:
updated in https://github.com/elastic/kibana/pull/179547/commits/f4cd520e2c085fb78f52bdd4daeea5caf06b7c0c